src/test/java/org/xstefank/check/SkipCheckTest.java

```diff
@@ -0,0 +1,122 @@
+    }
+
+    @Test
+    public void testSkipByTitleRegexMatch(){
+        testTitle.setPattern(Pattern.compile("test title"));
+        testSkipPatterns.setTitle(testTitle);
+        testFormatConfig = setUpFormatConfig(testSkipPatterns);
+
+        Assert.assertTrue("Method skipByTitle cannot match valid regex", SkipCheck.shouldSkip(testPullRequestPayload, testFormatConfig));
+    }
+
+    @Test
+    public void testSkipByTitleRegexNonMatch(){
+        testTitle.setPattern(Pattern.compile("can't.*match.*this"));
+        testSkipPatterns.setTitle(testTitle);
+        testFormatConfig = setUpFormatConfig(testSkipPatterns);
+
+        Assert.assertFalse("Method skipByTitle matched invalid regex",SkipCheck.shouldSkip(testPullRequestPayload, testFormatConfig));
+    }
+
+    @Test
+    public void testSkipByCommitRegexMatch(){
+        suppress(method(GitHubAPI.class, TestUtilities.getSuppressMethodName()));
+
+        testCommit.setPattern(Pattern.compile("test commit"));
+        testSkipPatterns.setCommit(testCommit);
+        testFormatConfig = setUpFormatConfig(testSkipPatterns);
+
+        PowerMockito.stub(PowerMockito.method(GitHubAPI.class, TestUtilities.getStubMethodName(), JsonNode.class)).toReturn(testCommitsPayload);
+
+        Assert.assertTrue("Method skipByCommit cannot match valid regex",SkipCheck.shouldSkip(testPullRequestPayload, testFormatConfig));
+    }
+
+    @Test
+    public void testSkipByCommitRegexNonMatch(){
+        suppress(method(GitHubAPI.class, TestUtilities.getSuppressMethodName()));
+
+        testCommit.setPattern(Pattern.compile("can't.*match.*this"));
+        testSkipPatterns.setCommit(testCommit);
+        testFormatConfig = setUpFormatConfig(testSkipPatterns);
+
+        PowerMockito.stub(PowerMockito.method(GitHubAPI.class, TestUtilities.getStubMethodName(), JsonNode.class)).toReturn(testCommitsPayload);
+
+        Assert.assertFalse("Method skipByCommit matched invalid regex",SkipCheck.shouldSkip(testPullRequestPayload, testFormatConfig));
+    }
+
+    @Test
+    public void testSkipByPullRequestDescriptionRegexMatch(){
+        testDescription.setPattern(Pattern.compile("test description"));
+        testSkipPatterns.setDescription(testDescription);
+        testFormatConfig = setUpFormatConfig(testSkipPatterns);
+
+        Assert.assertTrue("Method skipByPullRequestDescription cannot match valid regex",SkipCheck.shouldSkip(testPullRequestPayload, testFormatConfig));
+    }
+
+    @Test
+    public void testSkipByPullRequestDescriptionRegexNonMatch(){
+        testDescription.setPattern(Pattern.compile("can't.*match.*this"));
+        testSkipPatterns.setDescription(testDescription);
+        testFormatConfig = setUpFormatConfig(testSkipPatterns);
+
+        Assert.assertFalse("Method skipByPullRequestDescription matched invalid regex",SkipCheck.shouldSkip(testPullRequestPayload, testFormatConfig));
+    }
+
+    @Test
+    public void testShouldSkipNullSkipPatterns() {
```

Comment:
you are not passing null anywhere in this method, default or empty would be better